src/ServiceManagement/Compute/Commands.ServiceManagement/Common/DiagnosticsHelper.cs

```diff
@@ -0,0 +1,317 @@
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Azure.Common.Authentication.Models;
+using Microsoft.WindowsAzure.Commands.Common.Storage;
+using Microsoft.WindowsAzure.Management.Storage;
+using Microsoft.WindowsAzure.Management.Storage.Models;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Newtonsoft.Json;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.Common
+{
+    public static class DiagnosticsHelper
+    {
+        private static string XmlNamespace = "http://schemas.microsoft.com/ServiceHosting/2010/10/DiagnosticsConfiguration";
+        private static string EncodedXmlCfg = "xmlCfg";
+        private static string StorageAccount = "storageAccount";
+        private static string Path = "path";
+        private static string ExpandResourceDirectory = "expandResourceDirectory";
+        private static string LocalResourceDirectory = "localResourceDirectory";
+        private static string StorageAccountNameTag = "storageAccountName";
+        private static string StorageAccountKeyTag = "storageAccountKey";
+        private static string StorageAccountEndPointTag = "storageAccountEndPoint";
+
+        public static string DiagnosticsConfigurationElemStr = "DiagnosticsConfiguration";
+        public static string PublicConfigElemStr = "PublicConfig";
+        public static string PrivateConfigElemStr = "PrivateConfig";
+        public static string StorageAccountElemStr = "StorageAccount";
+        public static string PrivConfNameAttr = "name";
+        public static string PrivConfKeyAttr = "key";
+        public static string PrivConfEndpointAttr = "endpoint";
+
+        public static string GetJsonSerializedPublicDiagnosticsConfigurationFromFile(string configurationPath,
+            string storageAccountName)
+        {
+            return
+                JsonConvert.SerializeObject(
+                    DiagnosticsHelper.GetPublicDiagnosticsConfigurationFromFile(configurationPath, storageAccountName));
+        }
+
+        public static Hashtable GetPublicDiagnosticsConfigurationFromFile(string configurationPath, string storageAccountName)
+        {
+            using (StreamReader reader = new StreamReader(configurationPath))
+            {
+                return GetPublicDiagnosticsConfiguration(reader.ReadToEnd(), storageAccountName);
+            }
+        }
+
+        public static Hashtable GetPublicDiagnosticsConfiguration(string config, string storageAccountName)
+        {
+            // find the <WadCfg> element and extract it
+            int wadCfgBeginIndex = config.IndexOf("<WadCfg>");
+            if (wadCfgBeginIndex == -1)
+            {
+                throw new ArgumentException("Cannot find the WadCfg element in the config.");
```

Comment:
Please add all strings for the exceptions to the resources and reference it from there.